iam.tf

```
@@ -1,5 +1,6 @@
 locals {
-  create_role = var.create && var.create_role
+  create_role           = var.create && var.create_role
+  create_role_for_pipes = var.create && var.create_role && var.create_pipes
```

should these be switched around a bit? 

```suggestion
  create_pipes          = var.create && var.create_pipes
  create_role_for_pipes = local.create_pipes && var.create_role
```